python/scripts/filter_osm.py

```diff
@@ -12,111 +12,132 @@ class MissingInputError(Exception):
     pass
 
 def display_help():
-    """Function to display usage information"""
-    print(f"Usage: {os.path.basename(__file__)} [tag] [input_file]")
-    print("Tag:")
-    print("  -h/--help              : Display this help message")
-    print("  -id                    : Filter geographic objects based on relation ID")
-    print("  -b                     : Filter geographic objects based on bounding box (with osmium)")
-    print("  -t [expression_file]   : Filter objects based on tags in expression_file")
-    print("Option:")
-    print("  -s                     : Specify strategy type (optional for: -id, -b)")
-
-# Function to check if the file has a valid extension
-def is_valid_extension(file):
+    """Function to display help information instead of parser's default"""
+    help_text = f"""Usage: {os.path.basename(__file__)} [tag] [input_file] [option]
+ Tag:
+    -h/--help            : Display this help message
+    id                   : Filter geographic objects based on relation ID
+    b                    : Filter geographic objects based on bounding box (with osmium)
+    f                    : Filter objects based on tags in expression_file
+ Option:
+    -c [bbox]            : Specify bouding box (required for 'b' tag)
+                        (Bounding box is specified directly or in config geojson file)
+    -e [expression_file] : Specify path to expression file (required for 't' tag)
+    -r [relation_id]     : Specify relation_id (required for 'id' tag)
+    -s [strategy]        : Specify strategy type (optional for: 'id', 'b')
+    -R                   : Omit referenced objects (optional for 'f' tag)"""
+    print(help_text)
+
+def is_valid_extension(file: str):
+    """Function to check if the file has a valid extension."""
     valid_extensions = ["osm", "osm.pbf", "osm.bz2"]
     return any(file.endswith(f".{ext}") for ext in valid_extensions)
 
-def check_strategy(strategy):
+def check_strategy(strategy: str | None):
```

Comment:
This function never returns false. It should return nothing instead. The usage of it within the if statement misleads that there is the possibility that it returns false.